pkg/server/metrics.go

```diff
@@ -0,0 +1,114 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Shopify/goose/srvutil"
+	"github.com/cds-snc/covid-alert-server/pkg/keyclaim"
+	"github.com/cds-snc/covid-alert-server/pkg/persistence"
+	"github.com/gorilla/mux"
+
+	"context"
+)
+
+const ISODATE string = "2006-01-02"
+
+func NewMetricsServlet(db persistence.Conn, auth keyclaim.Authenticator) srvutil.Servlet {
+	log(nil, nil).Info("registering metrics servlet")
+	return &metricsServlet{db: db, auth: auth}
+}
+
+type metricsServlet struct {
+	db   persistence.Conn
+	auth keyclaim.Authenticator
+}
+
+const DATEFORMAT string = "\\d{4,4}-\\d{2,2}-\\d{2,2}"
+func (m metricsServlet) RegisterRouting(r *mux.Router) {
+	log(nil, nil).Info("registering metrics route")
+	r.HandleFunc(fmt.Sprintf("/events/{startDate:%s}", DATEFORMAT), m.handleEventRequest)
+
+
+}
+
+func authorizeRequest(r *http.Request) error {
+
+	uname, pword, ok := r.BasicAuth()
+	if !ok {
+		return fmt.Errorf("basic auth required for access")
+	}
+
+	metricUsername, uok := os.LookupEnv("METRICS_USERNAME")
+	if !uok {
+		log(nil, nil).Panic("Metrics username not set")
```

Comment:
Does log.Panic also terminate flow? In other places the code terminates the service if it is missing https://github.com/cds-snc/covid-alert-server/blob/master/pkg/keyclaim/authenticator.go#L25

Reply:
Now checking this on app start